File: pyg-groupck/pyg-manager-web/src/main/webapp/js/service/brandService.js
//定义一个服务层service
app.service("brandService",function ($http) {

    //把发送请求方法全部抽取到服务层
    //查询所有品牌
    this.findAll = function () {
        return $http.get("../brand/findAll");
    };

    //分页查询
    this.findPage = function (page,rows) {
        return $http.get("../brand/findByPage/" + page + "/" + rows);
    };

    //保存
    this.add = function (entity) {
        return $http.post("../brand/add",entity);
    }

    //更新
    this.update = function (entity) {
        return $http.post("../brand/update",entity);
    };

    //定义根据id查询
    this.findOne = function (id) {
        return $http.get("../brand/findOne/" + id);
    };

    //删除方法
    this.dele = function (ids) {
        return $http.get("../brand/delete/" + ids);
    }

    //定义查询请求，查询品牌下拉列表，进行多项选择
    this.findBrandSelect2List = function () {
        return $http.get("../brand/findBrandSelect2List");
    };

});

File: pyg-groupck/pyg-manager/pyg-manager-interface/src/main/java/com/pyg/manager/service/BrandService.java
package com.pyg.manager.service;

import com.pyg.pojo.TbBrand;
import com.pyg.utils.PageResult;

import java.util.List;
import java.util.Map;

/**
 * Created by on 2018/8/10.
 */
public interface BrandService {
    /**
     * 需求：查询所有品牌数据
     */
    public List<TbBrand> findAll();

    /**
     * 需求：品牌分页查询展示
     * 参数：Integer page,Integer rows
     * 返回值：分页包装类对象
     */
    public PageResult findByPage(Integer page,Integer rows);

    /**
     * 需求：实现品牌数据添加
     */
    public void insert(TbBrand brand);

    /**
     * 需求：根据id查询品牌数据
     */
    public TbBrand findOne(Long id);

    /**
     * 需求：更新品牌数据
     */
    public void update(TbBrand brand);

    /**
     * 需求：删除品牌数据
     */
    public void delete(Long[] ids);

    /**
     * 定义查询请求，查询品牌下拉列表，进行多项选择
     */
    public List<Map> findBrandSelect2List();
}
